### Analisis exploratorio
Importo las librerias que voy a necesitar y cargo los dataset con los que voy a trabajar. Se trata de 3 dataset de tres áreas de una fábrica de hojas de papel. los datos se toman cada un minuto. Los dataset vienen tal como estan en la base de datos. Es decir sin limpiar.

Operaciones: Se encarga de la producción.\
Aditivos: Se encarga de la aplicación de productos químicos necesarios para la producción.\
Caliedad: Se encarga de controlar las variables de calidad importantes para los clientes.

In [502]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [503]:
df_operaciones = pd.read_csv('operaciones.csv')
df_aditivos = pd.read_csv('aditivos.csv')
df_calidad = pd.read_csv('calidad.csv')

C:\Users\sas\AppData\Local\Temp\ipykernel_27412\285350360.py:1: DtypeWarning: Columns (1,3,4,5,6,7,8,9,10,11,12,13) have mixed types. Specify dtype option on import or set low_memory=False.
  df_operaciones = pd.read_csv('operaciones.csv')
C:\Users\sas\AppData\Local\Temp\ipykernel_27412\285350360.py:2: DtypeWarning: Columns (1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18) have mixed types. Specify dtype option on import or set low_memory=False.
  df_aditivos = pd.read_csv('aditivos.csv')
C:\Users\sas\AppData\Local\Temp\ipykernel_27412\285350360.py:3: DtypeWarning: Columns (1,2,3,4,5,6,7,8,9,10,11,12,14,15,16,17) have mixed types. Specify dtype option on import or set low_memory=False.
  df_calidad = pd.read_csv('calidad.csv')


### Explorando los dataset
los datos estan como Dtype object por lo que tengo que convertirlos a su tipo de dato para explorarlos mejor.\
-Primero convierto la columna 'Date' de cada dataset a formato datetime. Dado que todas tiene el mismo formato.\
-El dataset operaciones tiene la columna 'Grado' categórica.\
-El dataset aditivos no tiene columnas categóricas.\
-El dataset calidad tiene la columna 'Brillo_75' categórica.


In [556]:
#df_operaciones.head()
#df_operaciones.describe()
df_operaciones.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 132481 entries, 0 to 132480
Data columns (total 14 columns):
 #   Column           Non-Null Count   Dtype         
---  ------           --------------   -----         
 0   Date             132481 non-null  datetime64[ns]
 1   Corte_hoja       132447 non-null  float64       
 2   Grado            132481 non-null  object        
 3   Velocidad_pope   132461 non-null  float64       
 4   Gramaje_cara     132273 non-null  float64       
 5   Gramaje_Reverso  132273 non-null  float64       
 6   Gramaje_Tripa    132273 non-null  float64       
 7   Gramaje_prote    132273 non-null  float64       
 8   Caudal_cara      132462 non-null  float64       
 9   Caudal_Prote     132462 non-null  float64       
 10  Caudal_Reverso   132462 non-null  float64       
 11  Caudal_Tripa     132462 non-null  float64       
 12  secado_1         132438 non-null  float64       
 13  secado_2         132438 non-null  float64       
dtypes: datetime64[ns](1)

In [506]:
#df_aditivos.head()
#df_aditivos.describe()
df_aditivos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 132481 entries, 0 to 132480
Data columns (total 19 columns):
 #   Column                    Non-Null Count   Dtype 
---  ------                    --------------   ----- 
 0   Date                      132481 non-null  object
 1   Resina_CARA               132481 non-null  object
 2   Resina_TRIPA              132481 non-null  object
 3   Resina_REVERSO            132481 non-null  object
 4   Agente_Retencion          132481 non-null  object
 5   Almidon_cationico_cara    132481 non-null  object
 6   Soda_CARA                 132481 non-null  object
 7   Soda_REVERSO              132481 non-null  object
 8   Agente_Drenaje            132481 non-null  object
 9   polymin_SK                132481 non-null  object
 10  Sulfato_CARA              132481 non-null  object
 11  Sulfato_TRIPA             132481 non-null  object
 12  Sulfato_REVERSO           132481 non-null  object
 13  Almidon_CARA-PROTECCION   132481 non-null  object
 14  Almi

In [507]:
#df_calidad.head()
#df_calidad.describe()
df_calidad.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 132481 entries, 0 to 132480
Data columns (total 18 columns):
 #   Column              Non-Null Count   Dtype 
---  ------              --------------   ----- 
 0   Date                132481 non-null  object
 1   Gramaje_pope        132481 non-null  object
 2   Humedad_pope        132481 non-null  object
 3   Espesor_pope        132481 non-null  object
 4   Rigidez_Long        132481 non-null  object
 5   Rigidez_Transv      132481 non-null  object
 6   Scott_Bond          132481 non-null  object
 7   Blancura_Cara       132481 non-null  object
 8   Blancura_Reverso    132481 non-null  object
 9   Cobb_Cara           132481 non-null  object
 10  Cobb_Cara_Base      132481 non-null  object
 11  Cobb_Tripa          132481 non-null  object
 12  Cobb_Reverso        132481 non-null  object
 13  Brillo_75           132481 non-null  object
 14  Lisura_PPS          132481 non-null  object
 15  Tersura_Sheffield   132481 non-null  object
 16  Bl

In [508]:
# Convierte las columnas de fecha que vienen en formato cadena de caracteres

df_operaciones['Date'] = pd.to_datetime(df_operaciones['Date'], format='%m/%d/%Y %H:%M:%S')
df_aditivos['Date'] = pd.to_datetime(df_aditivos['Date'], format='%m/%d/%Y %H:%M:%S')
df_calidad['Date'] = pd.to_datetime(df_calidad['Date'], format='%m/%d/%Y %H:%M:%S')

In [509]:
print(df_operaciones.columns)

Index(['Date', 'Corte_hoja', 'Grado', 'Velocidad_pope', 'Gramaje_cara',
       'Gramaje_Reverso', 'Gramaje_Tripa', 'Gramaje_prote', 'Caudal_cara',
       'Caudal_Prote', 'Caudal_Reverso', 'Caudal_Tripa', 'secado_1',
       'secado_2'],
      dtype='object')


In [510]:
# Convierte las columnas del dataset operaciones con datos float que vienen en formato cadena de caracteres

columnas_float_op =['Corte_hoja', 'Velocidad_pope', 'Gramaje_cara','Gramaje_Reverso', 'Gramaje_Tripa',
                 'Gramaje_prote', 'Caudal_cara', 'Caudal_Prote', 'Caudal_Reverso', 'Caudal_Tripa', 'secado_1',
                 'secado_2']

for columna in columnas_float_op:
    df_operaciones[columna] = pd.to_numeric(df_operaciones[columna], errors='coerce')
    

In [511]:
print(df_aditivos.columns)

Index(['Date', 'Resina_CARA', 'Resina_TRIPA', 'Resina_REVERSO',
       'Agente_Retencion', 'Almidon_cationico_cara', 'Soda_CARA ',
       'Soda_REVERSO', 'Agente_Drenaje', 'polymin_SK', 'Sulfato_CARA',
       'Sulfato_TRIPA', 'Sulfato_REVERSO', 'Almidon_CARA-PROTECCION',
       'Almidon_PROTECCION-TRIPA', 'Almidon_REVERSO', 'Encolante_PRE',
       'Encolante_CARA', 'Encolante_REVERSO'],
      dtype='object')


In [512]:
# Convierte las columnas del dataset aditivos con datos float que vienen en formato cadena de caracteres

columnas_float_ad = ['Resina_CARA', 'Resina_TRIPA', 'Resina_REVERSO',
       'Agente_Retencion', 'Almidon_cationico_cara', 'Soda_CARA ',
       'Soda_REVERSO', 'Agente_Drenaje', 'polymin_SK', 'Sulfato_CARA',
       'Sulfato_TRIPA', 'Sulfato_REVERSO', 'Almidon_CARA-PROTECCION',
       'Almidon_PROTECCION-TRIPA', 'Almidon_REVERSO', 'Encolante_PRE',
       'Encolante_CARA', 'Encolante_REVERSO']

for columna in columnas_float_ad:
    df_aditivos[columna] = pd.to_numeric(df_aditivos[columna], errors='coerce')

In [513]:
print(df_calidad.columns)

Index(['Date', 'Gramaje_pope', 'Humedad_pope', 'Espesor_pope', 'Rigidez_Long',
       'Rigidez_Transv', 'Scott_Bond', 'Blancura_Cara', 'Blancura_Reverso',
       'Cobb_Cara', 'Cobb_Cara_Base', 'Cobb_Tripa', 'Cobb_Reverso',
       'Brillo_75', 'Lisura_PPS', 'Tersura_Sheffield', 'Blistering_Cara',
       'Blistering_Reverso'],
      dtype='object')


In [514]:
# Convierte las columnas del dataset calidad con datos float que vienen en formato cadena de caracteres

columnas_float_ca = ['Gramaje_pope', 'Humedad_pope', 'Espesor_pope', 'Rigidez_Long',
       'Rigidez_Transv', 'Scott_Bond', 'Blancura_Cara', 'Blancura_Reverso',
       'Cobb_Cara', 'Cobb_Cara_Base', 'Cobb_Tripa', 'Cobb_Reverso', 'Lisura_PPS', 'Tersura_Sheffield', 'Blistering_Cara',
       'Blistering_Reverso']

for columna in columnas_float_ca:
    df_calidad[columna] = pd.to_numeric(df_calidad[columna], errors='coerce')

Guardo estos nuevos dataset como archivos csv para usarlos en etapas posteriores del proyecto

In [516]:
df_operaciones.to_csv('ds_operaciones.csv', index=False)
df_aditivos.to_csv('ds_aditivos.csv', index=False)
df_calidad.to_csv('ds_calidad.csv', index=False)

#### Los tres dataset tienen el mismo número de muestras (132481 filas)

In [518]:
print('Tamaño del dataset')
print(f' Cantidad de muestras del dataset de operaciones: ', len(df_operaciones['Date']))
print(f' Cantidad de muestras del dataset de aditivos: ', len(df_aditivos['Date']))
print(f' Cantidad de muestras del dataset de calidad: ', len(df_calidad['Date']))

Tamaño del dataset
 Cantidad de muestras del dataset de operaciones:  132481
 Cantidad de muestras del dataset de aditivos:  132481
 Cantidad de muestras del dataset de calidad:  132481


#### Los tres dataset comienzan al mismo tiempo 2024-05-01 00:00

In [520]:
print('Fecha de inicio')
print(f' Fecha de inicio dataset operaciones: ', df_operaciones['Date'].min())
print(f' Fecha de inicio dataset operaciones: ', df_aditivos['Date'].min())
print(f' Fecha de inicio dataset operaciones: ', df_calidad['Date'].min())

Fecha de inicio
 Fecha de inicio dataset operaciones:  2024-05-01 00:00:00
 Fecha de inicio dataset operaciones:  2024-05-01 00:00:00
 Fecha de inicio dataset operaciones:  2024-05-01 00:00:00


#### Los tres dataset finalizan al mismo tiempo 2024-08-01 00:00

In [522]:
print('Fecha de termino')
print(f' Fecha de termino dataset operaciones: ', df_operaciones['Date'].max())
print(f' Fecha de termino dataset operaciones: ', df_aditivos['Date'].max())
print(f' Fecha de termino dataset operaciones: ', df_calidad['Date'].max())

Fecha de termino
 Fecha de termino dataset operaciones:  2024-08-01 00:00:00
 Fecha de termino dataset operaciones:  2024-08-01 00:00:00
 Fecha de termino dataset operaciones:  2024-08-01 00:00:00


In [523]:
print('Datos no numéricos del dataset de operaciones')
for i in columnas_float_op:
    print(f' {i}: ', df_operaciones[i].isna().sum())

Datos no numéricos del dataset de operaciones
 Corte_hoja:  34
 Velocidad_pope:  20
 Gramaje_cara:  208
 Gramaje_Reverso:  208
 Gramaje_Tripa:  208
 Gramaje_prote:  208
 Caudal_cara:  19
 Caudal_Prote:  19
 Caudal_Reverso:  19
 Caudal_Tripa:  19
 secado_1:  43
 secado_2:  43


In [524]:
print('Datos no numéricos del dataset de aditivos')
for i in columnas_float_ad:
    print(f' {i}: ', df_aditivos[i].isna().sum())

Datos no numéricos del dataset de aditivos
 Resina_CARA:  34
 Resina_TRIPA:  34
 Resina_REVERSO:  34
 Agente_Retencion:  34
 Almidon_cationico_cara:  34
 Soda_CARA :  1153
 Soda_REVERSO:  3006
 Agente_Drenaje:  34
 polymin_SK:  34
 Sulfato_CARA:  36
 Sulfato_TRIPA:  34
 Sulfato_REVERSO:  34
 Almidon_CARA-PROTECCION:  34
 Almidon_PROTECCION-TRIPA:  34
 Almidon_REVERSO:  34
 Encolante_PRE:  43
 Encolante_CARA:  34
 Encolante_REVERSO:  43


#### Las columnas 'Encolante_PRE' 'Encolante_CARA' 'Encolante_REVERSO' tiene solo ceros y NaN

In [526]:
print(f'Valores únicos en Encolante_PRE ', np.unique(df_aditivos['Encolante_PRE']))
print(f'Valores únicos en Encolante_CARA ', np.unique(df_aditivos['Encolante_CARA']))
print(f'Valores únicos en Encolante_PRE ', np.unique(df_aditivos['Encolante_REVERSO']))

Valores únicos en Encolante_PRE  [ 0. nan]
Valores únicos en Encolante_CARA  [ 0. nan]
Valores únicos en Encolante_PRE  [ 0. nan]


#### En el dataset de calidad hay tres columnas que tienen un gran número de valores NaN con más de 30000

In [528]:
print('Datos no numéricos del dataset de calidad')
for i in columnas_float_ca:
    print(f' {i}: ', df_calidad[i].isna().sum())

Datos no numéricos del dataset de calidad
 Gramaje_pope:  1
 Humedad_pope:  1
 Espesor_pope:  1
 Rigidez_Long:  1127
 Rigidez_Transv:  1127
 Scott_Bond:  1122
 Blancura_Cara:  1127
 Blancura_Reverso:  30896
 Cobb_Cara:  1380
 Cobb_Cara_Base:  2570
 Cobb_Tripa:  1380
 Cobb_Reverso:  31130
 Lisura_PPS:  1127
 Tersura_Sheffield:  1127
 Blistering_Cara:  1251
 Blistering_Reverso:  30979


#### La columna 'Brillo_75' solo tiene dos valores y no son valores de calidad, por lo que pueden ser errores de medición

In [530]:
print(f'Valores unicos en Brillo_75 ',np.unique(df_calidad['Brillo_75']))

Valores unicos en Brillo_75  ['I/O Timeout' 'Intf Shut']


#### La columna categórica 'Grado' que contiene los tipos de fabricación tiene dos tipos de datos que no corresponden

In [532]:
print(np.unique(df_operaciones['Grado']))

['2C205B' '2C215M' '2C225B' '2C235M' '2C240B' '2C245C' '2C250M' '2C295C'
 '2C325M' '2F225M' '3C250B' '3C255C' '3C260M' '3C265C' '3C270B' '3C275M'
 '3C285C' '3C290B' '3C295C' '3C295M' '3C310C' '3C310M' '3C311M' '3C315B'
 '3C325M' '3C330C' '3C340B' '3C345C' '3C370B' '3C370C' '3C370M'
 'Cargando...' 'I/O Timeout']


In [566]:
df_operaciones['Grado'].value_counts()

Grado
3C270B         16065
2C245C         13680
3C265C         12940
2C225B          9674
3C330C          7935
3C250B          7478
3C285C          6589
3C290B          5701
3C340B          5482
2C205B          5430
3C260M          4627
2C250M          3783
3C310C          3698
3C370C          3686
3C255C          3464
3C295C          2666
2C240B          2462
3C295M          2433
2C235M          2425
3C345C          2356
3C315B          2346
2C215M          1972
3C275M          1673
3C370B          1269
2C325M           641
3C325M           556
2F225M           413
3C370M           371
3C310M           336
3C311M           167
2C295C           160
Cargando...        2
I/O Timeout        1
Name: count, dtype: int64

In [568]:
df_operaciones.describe()

Date     Corte_hoja  Velocidad_pope  \
count                         132481  132447.000000   132461.000000   
mean   2024-06-15 23:59:59.999999488       1.838665      646.782498   
min              2024-05-01 00:00:00       0.000000      -15.347968   
25%              2024-05-24 00:00:00       0.000000      614.886963   
50%              2024-06-16 00:00:00       0.000000      701.128967   
75%              2024-07-09 00:00:00       0.000000      764.339722   
max              2024-08-01 00:00:00      38.000000      835.518005   
std                              NaN       5.748362      190.041448   

       Gramaje_cara  Gramaje_Reverso  Gramaje_Tripa  Gramaje_prote  \
count  1.322730e+05     1.322730e+05   1.322730e+05   1.322730e+05   
mean   3.115721e+17     3.116696e+17   3.099060e+17  -1.298967e+17   
min    0.000000e+00    -1.419553e-02  -9.223231e+18  -9.223231e+18   
25%    2.447920e+01     1.124488e+02   2.807080e+01   2.423652e+01   
50%    2.581618e+01     1.340399e+02   2.931640e+01   2.566458e+01   
75%    2.679367e+01     1.604347e+02   3.043030e+01   2.817603e+01   
max    9.223231e+18     9.223231e+18   9.223231e+18   9.223231e+18   
std    1.665604e+18     1.665775e+18   1.663947e+18   1.684368e+18   

         Caudal_cara   Caudal_Prote  Caudal_Reverso   Caudal_Tripa  \
count  132462.000000  132462.000000   132462.000000  132462.000000   
mean    29780.939252   25552.342385    29587.195243   52721.982415   
min         0.000000       0.000000        0.000000       0.000000   
25%     31136.529297   26122.410156    30132.607422   53609.082031   
50%     31465.341797   26645.742188    30955.140625   55479.755859   
75%     31765.895020   27542.256836    31881.950195   57314.079102   
max     38455.640625   33526.417969    37356.953125   62762.140625   
std      7097.397193    5793.469157     6762.969841   12559.747962   

            secado_1       secado_2  
count  132438.000000  132438.000000  
mean       68.293929      14.462984  
min         0.000000       0.000000  
25%        62.800757      14.793885  
50%        69.293243      15.482213  
75%        78.064667      16.402333  
max       129.264511      25.928455  
std        19.087921       4.440533

In [570]:
df_aditivos.describe()

Date    Resina_CARA   Resina_TRIPA  \
count                         132481  132447.000000  132447.000000   
mean   2024-06-15 23:59:59.999999488      45.801763      14.753224   
min              2024-05-01 00:00:00      -1.934144      -8.833306   
25%              2024-05-24 00:00:00      41.571981       1.948116   
50%              2024-06-16 00:00:00      49.798222       3.911603   
75%              2024-07-09 00:00:00      54.943766      16.085376   
max              2024-08-01 00:00:00     139.779114     244.772827   
std                              NaN      15.388656      20.840228   

       Resina_REVERSO  Agente_Retencion  Almidon_cationico_cara  \
count   132447.000000     132447.000000           132447.000000   
mean        16.604986       1755.282630                5.157775   
min         -0.316548         -5.389500               -0.067399   
25%          1.886989       1762.378723                4.008838   
50%         13.170195       1858.142822                6.268974   
75%         29.337402       1969.515625                6.305811   
max        154.789062       2578.110596               28.439831   
std         15.797092        520.415195                1.659745   

          Soda_CARA    Soda_REVERSO  Agente_Drenaje     polymin_SK  \
count  131328.000000  129475.000000   132447.000000  132447.000000   
mean       63.464242       9.309134       61.170987       7.847885   
min        -0.114291      -1.162184       -0.036006      -0.103420   
25%        56.768564       0.137895        5.169252      -0.035459   
50%        69.682602       0.166216       64.924606      -0.030888   
75%        76.362720      24.937186      103.255386      -0.028221   
max        99.798828      62.570919      199.014877     309.260315   
std        21.566822      14.161783       47.504230      23.420495   

        Sulfato_CARA  Sulfato_TRIPA  Sulfato_REVERSO  Almidon_CARA-PROTECCION  \
count  132445.000000  132447.000000    132447.000000            132447.000000   
mean      114.028032     660.814916        81.034066                 2.685114   
min        -1.130441      -2.971973        -1.288062                -0.245716   
25%       103.171379     533.474792        75.040882                 2.023381   
50%       125.474960     693.440674        83.291786                 2.655278   
75%       134.009888     845.675446        95.700954                 3.203486   
max       514.953125    1182.484741       310.218750                 6.720557   
std        37.531297     257.376649        27.850075                 0.793454   

       Almidon_PROTECCION-TRIPA  Almidon_REVERSO  Encolante_PRE  \
count             132447.000000    132447.000000       132438.0   
mean                   3.516256         3.296657            0.0   
min                    0.001181        -0.118341            0.0   
25%                    3.296180         2.815305            0.0   
50%                    3.617806         3.267084            0.0   
75%                    3.890093         3.860565            0.0   
max                    8.610660         5.711015            0.0   
std                    0.618996         0.815214            0.0   

       Encolante_CARA  Encolante_REVERSO  
count        132447.0           132438.0  
mean              0.0                0.0  
min               0.0                0.0  
25%               0.0                0.0  
50%               0.0                0.0  
75%               0.0                0.0  
max               0.0                0.0  
std               0.0                0.0

In [572]:
df_calidad.describe()

Date   Gramaje_pope   Humedad_pope  \
count                         132481  132480.000000  132480.000000   
mean   2024-06-15 23:59:59.999999488     270.976201       7.858187   
min              2024-05-01 00:00:00     164.487793       2.285179   
25%              2024-05-24 00:00:00     242.954399       7.483843   
50%              2024-06-16 00:00:00     263.551712       7.768088   
75%              2024-07-09 00:00:00     290.918694       8.178615   
max              2024-08-01 00:00:00     413.506012      13.455002   
std                              NaN      39.790753       0.628869   

        Espesor_pope   Rigidez_Long  Rigidez_Transv     Scott_Bond  \
count  132480.000000  131354.000000   131354.000000  131359.000000   
mean      475.286874      24.915437        8.446281     165.627993   
min       291.167389       6.400000        2.200000      76.199997   
25%       409.524864      17.100000        5.800000     145.199997   
50%       461.382126      21.900000        7.400000     168.699997   
75%       527.721436      31.100000       10.000000     185.100006   
max       726.740723      62.700001       22.100000     485.299988   
std        93.847241      11.830082        3.997060      28.348868   

       Blancura_Cara  Blancura_Reverso      Cobb_Cara  Cobb_Cara_Base  \
count  131354.000000     101585.000000  131101.000000   129911.000000   
mean       84.362094         77.389236       0.363978        0.597483   
min        70.099998         68.199997       0.280000        0.310000   
25%        81.599998         69.500000       0.340000        0.480000   
50%        84.500000         82.400002       0.360000        0.540000   
75%        87.099998         83.599998       0.380000        0.670000   
max        89.099998         87.099998       0.610000        1.330000   
std         2.801653          6.861847       0.031931        0.181527   

          Cobb_Tripa   Cobb_Reverso     Lisura_PPS  Tersura_Sheffield  \
count  131101.000000  101351.000000  131354.000000      131354.000000   
mean        0.368801       0.397944       1.304009          45.573397   
min         0.240000       0.290000       0.820000           0.000000   
25%         0.320000       0.370000       1.220000          31.000000   
50%         0.360000       0.400000       1.300000          45.000000   
75%         0.400000       0.410000       1.380000          60.000000   
max         0.860000       0.670000       2.920000         205.000000   
std         0.062882       0.033032       0.131325          21.349416   

       Blistering_Cara  Blistering_Reverso  
count    131230.000000       101502.000000  
mean         74.993157           77.742778  
min          38.799999           37.000000  
25%          69.599998           62.000000  
50%          75.199997           83.599998  
75%          80.000000           97.199997  
max         100.000000          320.000000  
std           7.811020           21.827760